### PR TITLE
slack: init at 2.0.1

### PR DESCRIPTION
**Pull request by greyson**
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [ X ] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ X ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

Fixes issue #<insert id>

cc @<maintainer>

---

_Please note, that points are not mandatory, but rather desired._

### CODE REVIEW

**Comment by grahamc** [comment]
This is unrelated, but I tried building and got:

```
curl: (22) The requested URL returned error: 503 Service Unavailable
/nix/store/wfh61l85g730j6l1qg58i9bfmsy5jdlh-xz-5.2.2/bin/xz: (stdin): File format not recognized
error: unexpected end-of-file
download of ‘https://cache.nixos.org/nar/1l0vpnwg7bjwbj7ghgd0v52w2d5bfagdjrdbh101dfyzvbmkdyyq.nar.xz’ failed: No such file or directory
could not download ‘/nix/store/y5j91bndalrgf7idifrzcqw24pjd45f5-spidermonkey-17.0.0’ from any binary cache
fetching path ‘/nix/store/y5j91bndalrgf7idifrzcqw24pjd45f5-spidermonkey-17.0.0’ failed with exit code 1
killing process 12590
killing process 12589
error: some substitutes for the outputs of derivation ‘/nix/store/cf01mahy0a10xjzkj2b7ix9p71cif24f-gconf-2.32.4.drv’ failed (usually happens due to networking issues); try ‘--fallback’ to build derivation from source 
```

**Comment by grahamc** [comment]
When I cherry-picked to 16.03 (where I want to use it) and rebuilt, I get:

```
rmdir: failed to remove '/nix/store/191wxvaimw85glsky9xpyh6ddlaxwz5a-slack-2.0.1/usr': Directory not empty
builder for ‘/nix/store/z617zbw317qaqvyi1y0m321alkgm9pcc-slack-2.0.1.drv’ failed with exit code 1
error: build of ‘/nix/store/z617zbw317qaqvyi1y0m321alkgm9pcc-slack-2.0.1.drv’ failed
```

File: pkgs/applications/networking/instant-messengers/slack/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  inherit src;
+
+  buildInputs = [ dpkg ];
+  unpackPhase = "true";
+  buildCommand = ''
+    mkdir -p $out
+    dpkg -x $src $out
+    cp -av $out/usr/* $out
+    rmdir $out/usr
```
Comment (grahamc) [inline pkgs/applications/networking/instant-messengers/slack/default.nix]: I changed the rmdir to rm -rf and then the result works perfectly. Currently though, it does not build.

:+1:

**Comment by jagajaga** [comment]
cc @volhovM

File: pkgs/applications/networking/instant-messengers/slack/default.nix (hunk 1 above)
Comment (jagajaga) [inline pkgs/applications/networking/instant-messengers/slack/default.nix]: `rmdir: failed to remove '/nix/store/w74yx9yxqrs852bykl398ynwas4vhs1f-slack-2.0.1/usr': Directory not empty`
Same problem.

**Comment by jagajaga** [comment]
Pushed in https://github.com/NixOS/nixpkgs/commit/7fe888e521fc523b77b1a4208b12c858d127e457

**Comment by grahamc** [comment]
Yay! Thank you. Since this is so small but many users will want it, can you backport to 16.03?

**Comment by jagajaga** [comment]
@grahamc AFAIK we do backport updates to pkgs that were already in `16.03` there. If you need to use it then use unstable channel.

**Comment by grahamc** [comment]
My understanding is the policies aren't so well defined or strictly enforced. I've counted many people in #nixos on Freenode wanting to use and bundle slack, and until recently was broken. Other than policy, many users will be happy for this. Other than policy, is there a reason to _not_ backport it to the next release? This isn't a small program few people use.

**Comment by jagajaga** [comment]
That will make no difference between unstable and 16.03 then.

**Comment by grahamc** [comment]
Well OK, but note I disagree. I would prefer a user have the common packages they need much closer to them, than require them learn how to get unstable but retain a strict difference between 16.03 and master. I can understand the spirit of what you say, though and will let it drop.

**Comment by greyson** [comment]
So, sorry I didn't get back before now (eastern time, had work).  I had switched `cp` to `mv` but I think I accidentally undid that just before committing.  Should I resubmit a PR with that rebased on, submit a PR with only that change (how to comment, then), or just leave it as is?

**Comment by jagajaga** [comment]
@greyson I've done everything.

**Pull request closed without merging**